src/azure-cli/azure/cli/command_modules/resource/_params.py

```diff
@@ -399,6 +399,11 @@ def load_arguments(self, _):
     with self.argument_context('tag') as c:
         c.argument('tag_name', options_list=['--name', '-n'])
         c.argument('tag_value', options_list='--value')
+        c.argument('resource_id', options_list='--resource-id',
+                   help='The resource identifier for the tagged entity. A resource, a resource group or a subscription may be tagged.')
```

Comment:
> A resource, a resource group or a subscription may be tagged.

How about using `--scope`? Because the binding object is not only resource

Reply:
We prefer to use `--resource-id`. Considered `scope` and had a discussion within the team when worked on powershell cmdlet. `scope` could be much wider, since `subscription` and `resource group` are also resources (proxy resource).

Reply:
OK